### ISSUE TITLE
Create the design system pages

### ISSUE THREAD

**Issue by Aveline-art** [hackforla/website#1927]
### Overview

Jumping off of discussion that started in #1850, this will be the issue to discuss the specifics of the Design System webpages that will exist on our website.

Last comment on the [Design System webpages from #1850 ](https://github.com/hackforla/website/issues/1850#issuecomment-877859115).

### Action items

- [x] Decide on goals of the pages
- [x] Decide on design of the architecture
- [x] Create prototype of the design system
- [x] Open a discuss on prototype if needed
- [ ] ~Finalize design system pages~ - this will need to be a new issue (which has as its dependency, the design system team making new mockups for development).  See comment below

### Resources/Instructions
#577
#1416
#1850
[Figma](https://www.figma.com/file/0RRPy1Ph7HafI3qOITg0Mr/Hack-for-LA-Website?node-id=3464%3A3)

**Comment by Aveline-art** [comment]
MVP for prototype:

- [ ] Have all the information on the page
- [ ] Implement it in a way that is sustainable (meaning that it is easy to maintain and add more components in the future)
- [ ] Prototype will only be buttons and page cards
  - [ ] Sidenav should work such that clicking on the links will direct the user to the right pages/sections of the page

**Comment by adam-abundis** [comment]
### Notes from Design System Team Meeting - July 14, 2020

1. [Figma Design for Design System Website](https://www.figma.com/file/0RRPy1Ph7HafI3qOITg0Mr/Hack-for-LA-Website?node-id=3464%3A3) content is all there. @daniellex0 needs to add space (margin).

2. Current Structure of the Figma design
```
Buttons
  Sizes (small, medium, Large, X-Large)
  colors (primary, dark)

Page Cards
  sizes (small, medium, Large, X-Large)
  colors (primary, dark)

Other Cards

Input fields

Tags

Accordion

Modal

Dropdown

Side Nav
```

3. @abuna1985 did some research on design system templates and found:
    - [A Brad Frost design template in Jekyll repo](https://github.com/bradfrost/style-guide-guide) and the [example Jekyll site](http://bradfrost.github.io/style-guide-guide) for reference.
    - [A forked design template in Jekyll repo](https://github.com/lundegaard/design-system-template) and the [another example Jekyll site](https://lundegaard.github.io/design-system-template/) for reference

4. @daniellex0 MVP/Prototype goals
    - Used as a reference for developers
    - Have all the info easily available
    - Implement the site it so it is sustainable meaning:
        -  It is easy to maintain
        - Can easily add more elements in the future.
        
5. Other Questions?
   - Sidebar navigation - will it be static or are we going to have Jekyll build it?
   - Instead of saying and typing `design system` a bunch of times, maybe we should come up with a name (like Hack for LA Angel)?
 
 6. @arghmatey volunteered to take on issue #1927 and build Button and Card page.
 
 7. @Aveline-art designated @abuna1985 to oversee #1850.

**Comment by arghmatey** [comment]
Progress: Front end basics complete for page. Backend architecture is in progress.
Blockers: Became unexpectedly busy this week. 
Availability: Freeing up 7/31 and the week of 8/2
ETA: For completion - unsure. For good progress - early next week.

Edit: This comment was meant to be left on 7/30, but internet was constantly being dropped that day and did not post.

**Comment by Aveline-art** [comment]
@arghmatey You probably already know this, but I want to give you a heads up: the copy button is already a component on the website, and we are planning to make a issue to componentize it. You can feel free to put a placeholder for your template!

On that note, Bonnie also mentioned that, for MVP, there is no need to worry too much about content or styles as long as the overall object placement of the template fits.

**Comment by arghmatey** [comment]
Created draft pull request #2070 to keep track of progress.

1. Progress: Created front end and back end. Continuing research on best practices. Styling and adding the rest of component information to files still needs to be completed.
2. Blockers: Time! Busier than I expected, but I believe the hardest parts are done.
3. Availability: Monday evening, Tuesday afternoon
4. ETA: TBD - I would like to mention during Tuesday developers meeting and collect feedback.
5. Pictures: See draft pull request.

**Comment by arghmatey** [comment]
Progress: Huge progress has been made since last update. 
Blockers: Margins in Figma file will be changed. Image page cards have no standard component created.
Availability: Will hop back on when margins are fixed. 
ETA: Dependent on blockers. Since this is only the prototype, I believe we can move on without the image page cards. 
Pictures: See attached PR draft.

**Comment by arghmatey** [comment]
Progress: PR is submitted and waiting on reviews!
Blockers: N/A
Availability: Sunday AM
ETA: N/A
Pictures: See PR

**Comment by ExperimentsInHonesty** [comment]
@arghmatey what is the url for the design system?

**Comment by arghmatey** [comment]
@ExperimentsInHonesty https://www.hackforla.org/design-system

**Comment by ExperimentsInHonesty** [comment]
We are marking this issue closed

The current layout of the design system pages does not conform to the design system.
- [ ] ~Finalize design system pages~ - this will need to be a new issue (which has as its dependency, the design system team making new mockups for development).
